pkg/gadgets/interface.go

```diff
@@ -87,6 +87,8 @@ type Resolver interface {
 
 	// Unsubscribe undoes a previous call to Subscribe
 	Unsubscribe(key interface{})
+
+	PublishEvent(tracerID string, line string) error
```

Comment:
nit: `type Resolver interface` doesn't seem to be a right name after this change. We have to find something better but I have no ideas right now.